packages/tpetra/core/test/Map/Map_getMinGlobalIndex.cpp

```diff
@@ -0,0 +1,212 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+*/
+
+// Some Macro Magic to ensure that if CUDA and KokkosCompat is enabled
+// only the .cu version of this file is actually compiled
+#include <Tpetra_ConfigDefs.hpp>
+#include <Tpetra_Map.hpp>
+#include <Tpetra_TestingUtilities.hpp>
+#include <type_traits> // std::is_same
+
+// FINISH: add testing of operator==, operator!=, operator=, copy construct
+// put these into test_same_as and test_is_compatible
+
+namespace {
+
+  using Teuchos::null;
+
+  using Tpetra::TestingUtilities::getNode;
+  using Tpetra::TestingUtilities::getDefaultComm;
+
+  using Tpetra::createUniformContigMapWithNode;
+
+  using Teuchos::Array;
+  using Teuchos::ArrayView;
+  using Teuchos::as;
+  using Teuchos::RCP;
+  using Teuchos::arcp;
+  using Teuchos::rcp;
+  using Teuchos::outArg;
+  using Teuchos::Tuple;
+  using Teuchos::tuple;
+  using Tpetra::Map;
+  using Tpetra::global_size_t;
+  using Tpetra::DefaultPlatform;
+  using std::sort;
+  using std::find;
+  using Teuchos::broadcast;
+  using Teuchos::OrdinalTraits;
+  using Teuchos::Comm;
+  using std::endl;
+
+  ////
+  TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( Map, getMinGlobalIndex_nonuniform, LO, GO )
+  {
+    typedef Map<LO,GO> Map;
+
+    out << "Test: Map, getMinGlobalIndex" << std::endl;
+    Teuchos::OSTab tab0 (out);
+
+    // create a comm
+    RCP<const Comm<int> > comm = getDefaultComm();
+    const int numRanks = comm->getSize();
+    const int myRank = comm->getRank();
+    // create a contiguous uniform distributed map with numLocal entries per node
+    const size_t        numLocalRef = 5;
+    const size_t        numLocal    = (myRank == 0) ? 0 : numLocalRef;
+    const global_size_t numGlobal   = (numRanks - 1)*numLocalRef;
+    const GO indexBase = 0;
+    //
+
+    Map map (numGlobal, numLocal, indexBase, comm);
+    // create data to check validity of the map
+    const GO myMinGID =
+      (myRank == 0) ? std::numeric_limits<GO>::max() : (myRank - 1)*numLocal;
+    const GO myMaxGID =
+      (myRank == 0) ? std::numeric_limits<GO>::lowest() : myRank*numLocal - 1;
+    GO minAllGIDs, maxAllGIDs;
+    Teuchos::reduceAll<LO, GO>(*comm, Teuchos::REDUCE_MIN, myMinGID, outArg(minAllGIDs));
```

Comment:
The first template parameter of `Teuchos::reduceAll` should always be `int`, to match the template parameter of `Teuchos::Comm`.

Reply:
OK, switched template parameter to `int`.